lib/board.rb

```diff
@@ -0,0 +1,61 @@
+require 'pry'
+class Board
+
+
+	attr_accessor :cells, :edges, :corners, :center
+
+	def initialize
+		@cells = Array.new(9," ")
+		@edges = [2,4,6,8]
+		@corners= [1,3,7,9]
+		@center= [5]
+
+	end
+
+	def reset!
+		@cells.clear
+		@cells = Array.new(9," ")
+	end
+
+	def display
+		puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+	    puts "-----------"
+	    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+	    puts "-----------"
+	    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+	end
+
+	def position(position)
+		user_position = @cells[position.to_i-1]
+	end
+
+	def full?
+		@cells.all? {|cell| cell == "X" || cell == "O" }
+	end
+
+	def turn_count
+		turns = @cells.select {|cell| cell == "X" || cell == "O"}
+		turns.size
+	end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  @cells.count {|cell| cell == "X" || cell == "O"}
end
```